webview/src/plots/hooks/useGetPlot.ts

```diff
@@ -1,40 +1,37 @@
-import { CustomPlotData, PlotsSection } from 'dvc/src/plots/webview/contract'
-import { SpecWithTitles } from 'dvc/src/plots/vega/util'
+import { PlotsSection } from 'dvc/src/plots/webview/contract'
 import { useCallback, useEffect, useState } from 'react'
 import { useSelector } from 'react-redux'
-import { PlainObject } from 'react-vega'
+import { VisualizationSpec } from 'react-vega'
 import { plotDataStore } from '../components/plotDataStore'
 import { PlotsState } from '../store'
+import { fillTemplate } from '../components/vegaLite/util'
 
-export const useGetPlot = (section: PlotsSection, id: string) => {
-  const isTemplatePlot = section === PlotsSection.TEMPLATE_PLOTS
-  const storeSection = isTemplatePlot ? 'template' : 'custom'
-  const snapshot = useSelector(
-    (state: PlotsState) => state[storeSection].plotsSnapshots
-  )
+export const useGetPlot = (
```

Comment:
[F] I decided to split out getting the titles into a different hook. LMK if you want me to bring them back together.